docs/Makefile.am

```diff
@@ -889,4 +889,13 @@ man1_MANS += \
 man3_MANS += $(OSHMEM_MAN3_BUILT)
 endif
 
+install-data-hook:
+	$(MKDIR_P) $(DESTDIR)$(docdir)
+	cp -r $(srcdir)/_build/html $(DESTDIR)$(docdir)
+	find $(DESTDIR)$(docdir) -type d -exec chmod 0755 {} \;
```

Comment:
First, there needs to be a comment about this behavior, because the history will be lost on why we're chmodding if the only place the history exists is a comment in the PR.

Second, I don't think this is actually Automake's behavior.  I agree AM installs files as 0644, but it appears that there's more logic around setting directory permissions, which I haven't entirely pulled apart.